(globalThis["webpackChunkquasar_tufte"]=globalThis["webpackChunkquasar_tufte"]||[]).push([[704],{32704:(e,t,a)=>{"use strict";a.r(t),a.d(t,{default:()=>Ce});var n=a(59835),o=a(86970),i=a(26174),r=a.n(i),s=a(73578),l=a.n(s),d=a(85879),c=a.n(d),u=a(22903),h=a.n(u),m=a(30287),f=a.n(m),p=a(73014),g=a.n(p);const b=e=>((0,n.dD)("data-v-0a4a9b2c"),e=e(),(0,n.Cn)(),e),w={class:"tufte-body"},k=(0,n.Uk)(),D=(0,n.Uk)(),y=(0,n.Uk)("\nLate 2023 the Storage Networking Industry Association (SNIA) together with the DNA Data Storage Alliance released their specification for what would be considered the master boot record but for a Deoxyribonucleic Acid (DNA) Storage Archive. In this post, we will briefly look at the specification and illustrate the format in action. They are released under the names Sector Zero "),A=(0,n.Uk)(" and Sector One "),v=(0,n.Uk)(". Sector Zero first is very short and just answers who (as in which vendor's technology) and how (as in which encoding/codec) was used to create the DNA archive. Sector One based on the previous information allows accessing some standardized or common metadata (reserved keywords) as well as optional vendor-specific information on the archive.\n"),S=(0,n.Uk)(),N=b((()=>(0,n._)("h2",null,"DNA Storage Fundamentals",-1))),U=(0,n.Uk)(),_=(0,n.Uk)("   \nTo provide some context on the basics to make sense of the specification rationals, let's briefly look at the fundamental building blocks and procedures that enable DNA storage. \nInformation in a DNA based data storage system are represented using four particular types of molecules, the (oligo)nucleotides or bases.\nIn nature, generally only four of these nucleotides are found in DNA which are called Adenin (A), Thymien (T), Cytosin (C) and Guianin (G).\nThese oligonucleatides can form chains that are linked together by a phosphodiester bond see "),x=(0,n.Uk)(".\nAdenine and Thymine form a complementary pair because they have two free hydrogen bonds.\nSimilarly, Cytosin and Guanine are complementary but with three free hydrogen bonds.\nThis property gives DNA a lot of it's stability and intrinsic redundancy, as a sequence, and its inverse can form the famous double helix which is remarkably resilient to outside disturbance.\n"),I=(0,n.Uk)(),W=b((()=>(0,n._)("img",{style:{width:"500px"},src:r()},null,-1))),O=(0,n.Uk)(),q=(0,n.Uk)("Illustration of the DNA double helix and the four bases Adenin and Thymine as well as Cytosin and Guinine that pair up due to the 2 and 3 hydrogen bonds respectively. Figure adapted from"),T=(0,n.Uk)("."),Z=(0,n.Uk)(),C=(0,n.Uk)("   \nWriting and reading DNA to encode information requires a number of procedures for which the full data life-cycle is illustrated in "),B=(0,n.Uk)(". Digital data is converted to an encoding scheme that maps to the four bases and accounts for different constraints imposed by synthesis, storage and sequencing technologies. In particular, the addition of primers for targeted retrieval of specific data fragments encoded in DNA and data redundancy to allow error correction of unavoidable errors both introduced during synthesis and in the sequencing process.\n"),j=(0,n.Uk)(),F=b((()=>(0,n._)("img",{src:l()},null,-1))),z=(0,n.Uk)(),L=(0,n.Uk)("Overview of the data life cycle for DNA storage to go back and forth between digital binary representations and DNA representations."),M=(0,n.Uk)(),P=b((()=>(0,n._)("p",null,"\nAs there are a wide spectrum of possible strategies to encode and organize information in DNA sequences, Sector Zero and Sector One aim to provide a common standard and a predictable location to describe and inquire about the properties and metadata of a given DNA archive.\nIn the following we look at the representation at the DNA level, even though in practice they would in many cases be also replicated in digital form such as an QR code, an EEPROM, or more centralized databases or indexes.\n  ",-1))),E=(0,n.Uk)(),G=b((()=>(0,n._)("h2",null,"Sector Zero: Who wrote and how?",-1))),J=(0,n.Uk)(),K=(0,n.Uk)("\nSector Zero is very basic encoding only two information: 1) a vendor ID and 2) a codec ID. This serves to facilitate the decoding of the more detailed metadata about the archive found in Sector One.\nIn line with current synthesis and sequencing capabilities the standard assumes to only have reliable reads and writes of about 150 bases long. \n"),R=(0,n.Uk)(' illustrates the breakdown of bases including amplification, sector identification primers, and the actual payload, dedicating 35 bases each for Vendor Id and Codec ID.\nThe standard defines four "reserved" sequences for the forward/reverse sector zero identification primers and specific sequences for unregistered vendor and codec IDs. \n  '),V=(0,n.Uk)(),H=b((()=>(0,n._)("img",{src:c()},null,-1))),Q=(0,n.Uk)(),X=(0,n.Uk)("Structure of Sector Zero used to reveal archive who created the archive and which codec was used. Here, using the reserved sequences for unregistered vendors and unregistered codecs."),Y=(0,n.Uk)(),$=b((()=>(0,n._)("h2",null,"Sector One: Various Additional Metadata",-1))),ee=(0,n.Uk)(),te=(0,n.Uk)("\nUnlike Sector Zero which is self-contained in a 150 base long sequence, Sector One needs to be distributing information across multiple sequences as illustrated in "),ae=(0,n.Uk)(". Again, 150 bases is an artifact of limitations to synthesis and sequencing capabilities, which might increase in the future. "),ne=(0,n.Uk)(),oe=b((()=>(0,n._)("img",{src:h()},null,-1))),ie=(0,n.Uk)(),re=(0,n.Uk)("Structure of Sector One."),se=(0,n.Uk)(),le=(0,n.Uk)("\nExcept for the dedicated amplification primers the codec is free to use the payload sections as it sees fit, but typically it will include a adressing scheme, a payload section and some redundancy to allow error correction as illustrated in "),de=(0,n.Uk)(".\n"),ce=(0,n.Uk)(),ue=b((()=>(0,n._)("img",{src:f()},null,-1))),he=(0,n.Uk)(),me=(0,n.Uk)("Structure of Sector One with an example scheme for a codec that includes a address section, the codex payload and error correction."),fe=(0,n.Uk)(),pe=(0,n.Uk)("Sector One is supposed to return a minified JSON-style object with a few agreed on sections: At the top-level some basic metadata such a ID, a description and a timestep are included. Nested are then additional sections with details about used hash algorithms, sequencing details with oligo counts and lengths, as well as information about the codec and the vendor. Finally, a custom section allows to include additional custom metadata. An example is illustrated in "),ge=(0,n.Uk)("."),be=(0,n.Uk)(),we=b((()=>(0,n._)("img",{src:g()},null,-1))),ke=(0,n.Uk)(),De=(0,n.Uk)("Example of the minified JSON that is encoded across multiple sequences that compose Sector One."),ye=(0,n.Uk)(),Ae=b((()=>(0,n._)("p",null,null,-1))),ve=(0,n.Uk)(),Se={key:0},Ne=b((()=>(0,n._)("h3",{class:"no-count"},"References",-1))),Ue=(0,n.Uk)(),_e={class:"fullwidth"};function xe(e,t,a,i,r,s){const l=(0,n.up)("blog-metadata"),d=(0,n.up)("t-cite"),c=(0,n.up)("t-ref"),u=(0,n.up)("center"),h=(0,n.up)("t-figure"),m=(0,n.up)("TufteFigure"),f=(0,n.up)("BibtexBibliography");return(0,n.wg)(),(0,n.iD)("div",w,[(0,n._)("article",null,[(0,n._)("section",null,[(0,n._)("h1",null,(0,o.zw)(e.metadata.title),1),k,(0,n.Wm)(l,{metadata:e.metadata},null,8,["metadata"]),D,(0,n._)("p",null,[y,(0,n.Wm)(d,{name:"DNADataStorage2023"}),A,(0,n.Wm)(d,{name:"DNADataStorage2023a"}),v])]),S,(0,n._)("section",null,[N,U,(0,n._)("p",null,[_,(0,n.Wm)(c,{label:"fig:DNA"}),x]),I,(0,n.Wm)(h,{label:"fig:DNA"},{caption:(0,n.w5)((()=>[q,(0,n.Wm)(d,{name:"FileDNAStructure2016"}),T])),default:(0,n.w5)((()=>[(0,n.Wm)(u,null,{default:(0,n.w5)((()=>[W])),_:1}),O])),_:1}),Z,(0,n._)("p",null,[C,(0,n.Wm)(c,{label:"fig:DNA Storage Model"}),B]),j,(0,n.Wm)(h,{label:"fig:DNA Storage Model"},{caption:(0,n.w5)((()=>[L])),default:(0,n.w5)((()=>[F,z])),_:1}),M,P]),E,(0,n._)("section",null,[G,J,(0,n._)("p",null,[K,(0,n.Wm)(c,{label:""}),R]),V,(0,n.Wm)(m,{fullwidth:"",label:"sector-zero"},{caption:(0,n.w5)((()=>[X])),default:(0,n.w5)((()=>[H,Q])),_:1})]),Y,(0,n._)("section",null,[$,ee,(0,n._)("p",null,[te,(0,n.Wm)(c,{label:"sector-one basic"}),ae]),ne,(0,n.Wm)(m,{fullwidth:"",label:"sector-one basic"},{caption:(0,n.w5)((()=>[re])),default:(0,n.w5)((()=>[oe,ie])),_:1}),se,(0,n._)("p",null,[le,(0,n.Wm)(c,{label:"sector-one codec"}),de]),ce,(0,n.Wm)(m,{fullwidth:"",label:"sector-one codec"},{caption:(0,n.w5)((()=>[me])),default:(0,n.w5)((()=>[ue,he])),_:1}),fe,(0,n._)("p",null,[pe,(0,n.Wm)(c,{label:"sector-one json"}),ge]),be,(0,n.Wm)(m,{fullwidth:"",label:"sector-one json"},{caption:(0,n.w5)((()=>[De])),default:(0,n.w5)((()=>[we,ke])),_:1}),ye,Ae]),ve,e.references?((0,n.wg)(),(0,n.iD)("section",Se,[Ne,Ue,(0,n._)("p",_e,[(0,n.Wm)(f,{bibtex:e.references},null,8,["bibtex"])])])):(0,n.kq)("",!0)])])}var Ie=a(60499),We={title:"A Brief Look at Sector Zero and Sector One for DNA Data Storage",authors:["Jakob Luettgau"],datetime:"2024-05-19T20:35:33Z",text:"Late 2023 the Storage Networking Industry Association (SNIA) together with the DNA Data Storage Alliance released their specification for what would be considered the master boot record but for a DNA Storage Archive. In this post, we will briefly look at the specification and illustrate the format in action.",image:"./assets/png/sector-one_codec.png",caption:null,published:!0},Oe=a(42426).Z;const qe=(0,n.aZ)({name:"BlogPost",components:{},setup(){const e=(0,Ie.iH)(Oe);return{metadata:We,references:e}}});var Te=a(11639);const Ze=(0,Te.Z)(qe,[["render",xe],["__scopeId","data-v-0a4a9b2c"]]),Ce=Ze},42426:(e,t,a)=>{"use strict";a.d(t,{Z:()=>n});const n="@article{10.1109/MC.2023.3272188,\n  title = {The {{DNA Data Storage Model}}},\n  author = {Landsman, Dave and Strauss, Karin},\n  year = {2023},\n  month = jul,\n  journal = {Computer},\n  volume = {56},\n  number = {7},\n  pages = {78--85},\n  issn = {0018-9162, 1558-0814},\n  doi = {10.1109/MC.2023.3272188},\n  url = {https://ieeexplore.ieee.org/document/10154188/},\n  urldate = {2023-12-11},\n  langid = {english}\n}\n\n@misc{DNADataStorage2023,\n  title = {{{DNA Data Storage Sector One}} v1.0.Pdf},\n  year = {2023},\n  month = nov,\n  publisher = {SNIA},\n  url = {https://www.snia.org/sites/default/files/technical-work/dna/release/DNA%20Data%20Storage%20Sector%20One%20v1.0.pdf},\n  urldate = {2024-03-19},\n  abstract = {The DNA Data Storage Sector One specification defines the recommended method and embodiment for storing archive metadata within a DNA data storage archive for the purpose of enabling an archive reader to read the archive and then consume the logical structure and its data contents.}\n}\n\n@misc{DNADataStorage2023a,\n  title = {{{DNA Data Storage Sector Zero}} v1.0.Pdf},\n  year = {2023},\n  month = nov,\n  publisher = {SNIA},\n  url = {https://www.snia.org/sites/default/files/technical-work/dna/release/DNA%20Data%20Storage%20Sector%20Zero%20v1.0.pdf},\n  urldate = {2024-03-19},\n  abstract = {The DNA Data Storage Sector Zero specification defines the recommended method and embodiment for storing basic vendor and CODEC information (sector zero contents) within a DNA data storage archive for the purpose of enabling an archive reader to then consume archive metadata (sector one) and data contents.}\n}\n\n@article{FileDNAStructure2016,\n  title = {File:{{DNA Structure}}+{{Key}}+{{Labelled}}.Pn {{NoBB}}.Png},\n  shorttitle = {File},\n  year = {2016},\n  month = jan,\n  journal = {Wikipedia},\n  url = {https://en.wikipedia.org/w/index.php?title=File:DNA_Structure%2BKey%2BLabelled.pn_NoBB.png&oldid=700345270},\n  urldate = {2024-04-19},\n  copyright = {Creative Commons Attribution-ShareAlike License},\n  langid = {english},\n  annotation = {Page Version ID: 700345270}\n}\n"},22903:(e,t,a)=>{e.exports=a.p+"img/sector-one_bare.25b2eb6f.png"},30287:(e,t,a)=>{e.exports=a.p+"img/sector-one_codec.cc9e33d5.png"},73014:(e,t,a)=>{e.exports=a.p+"img/sector-one_json.4d15ec5d.png"},85879:(e,t,a)=>{e.exports=a.p+"img/sector-zero.37ff3687.png"},73578:(e,t,a)=>{e.exports=a.p+"img/DNA-Storage-Model.f6ca8d9d.png"},26174:(e,t,a)=>{e.exports=a.p+"img/DNA.b48f0d63.svg"}}]);